### PR TITLE
✨ Source Ashby updates: new streams and extra fields for some existing streams

### LINKED ISSUE #32813: Source Ashby: more coverage of the relevant fields
### What area the feature impact?

Connectors

### Relevant Information

The [Ashby source](https://docs.airbyte.com/integrations/sources/ashby) is missing a number of fields from the Ashby API:

- [interview schedules](https://developers.ashbyhq.com/reference/interviewschedulelist)' interview events
- [interview plans](https://developers.ashbyhq.com/reference/interviewplanlist)
- [interviews](https://developers.ashbyhq.com/reference/interviewlist)
- [interview stages](https://developers.ashbyhq.com/reference/interviewstagelist)

... among others. (These seem to be the main ones in practice that limit the use of the connector at its current stage).

### PR DESCRIPTION
**Pull request by imrehg**
## What
These changes extend the Ashby source: new streams, and adding extra fields to some existing fields.

Also adding a configuration parameter that allows choosing whether to pull archived records / disabled users. There's a default value set that is consistent with the API's default behaviour (not pulling such records, users).

Closes airbytehq/airbyte#32813

## How
Adding new streams with their relevant schema, as well as expanding the most important incomplete endpoints with the missing schema values.

## Recommended reading order
1. `manifest.yaml`
2. schema JSONs
3. configured catalogue
4. docs

## 🚨 User Impact 🚨
*Are there any breaking changes? What is the end result perceived by the user?*

There should be no breaking change for the users. All fields are backwards compatible, and the new configuration options have sensible defaults.

*For connector PRs, use this section to explain which type of semantic versioning bump occurs as a result of the changes. Refer to our [Semantic Versioning for Connectors](https://docs.airbyte.com/contributing-to-airbyte/#semantic-versioning-for-connectors) guidelines for more information. **Breaking changes to connectors must be documented by an Airbyte engineer (PR author, or reviewer for community PRs) by using the [Breaking Change Release Playbook](https://docs.google.com/document/d/1VYQggHbL_PN0dDDu7rCyzBLGRtX-R3cpwXaY8QxEgzw/edit).***

The recommended semantic version is `0.2.0`, as this connector is likely still in it's early stage, but the changes expand the pulled data quite a bit.

*If there are breaking changes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.*

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- Connector version is set to `0.0.1`
    - `Dockerfile` has version `0.0.1`
- Documentation updated
    - Connector's `README.md`
    - Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - `docs/integrations/<source or destination>/<name>.md` including changelog with an entry for the initial version. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - `docs/integrations/README.md`

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Connector Generator</strong></summary>

- Issue acceptance criteria met
- PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/resources/pull-requests-handbook)
- If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:generateScaffolds` then checking in your changes
- Documentation which references the generator is updated as needed

</details>

<details><summary><strong>Updating the Python CDK</strong></summary>

### Airbyter

Before merging:
- Pull Request description explains what problem it is solving
- Code change is unit tested
- Build and my-py check pass
- Smoke test the change on at least one affected connector
   - On Github: Run [this workflow](https://github.com/airbytehq/airbyte/actions/workflows/connectors_tests.yml), passing `--use-local-cdk --name=source-<connector>` as options
   - Locally: `airbyte-ci connectors --use-local-cdk --name=source-<connector> test`
- PR is reviewed and approved
      
After merging:
- [Publish the CDK](https://github.com/airbytehq/airbyte/actions/workflows/publish-cdk-command-manually.yml)
   - The CDK does not follow proper semantic versioning. Choose minor if this the change has significant user impact or is a breaking change. Choose patch otherwise.
   - Write a thoughtful changelog message so we know what was updated.
- Merge the platform PR that was auto-created for updating the Connector Builder's CDK version
   - This step is optional if the change does not affect the connector builder or declarative connectors.

</details>

### CODE REVIEW

**Comment by imrehg** [comment]
With the acceptance tests I've found some more schema violations (from earlier, rather than these new streams), so I'm taking care of those ones shortly as well, hopefully as this PR.

**Comment by imrehg** [comment]
Hey :) I've updated the specs, all the relevant schema tests from `airbyte-ci connectors --name=source-ashby test` pass now.

Outstanding failure is due to "2 sequential full refresh". As I looked at the data from the two pulls, they are genuinely different and due to the upstream API. That's something I'll likely bring up with Ashby, though more debugging is needed. In practice, rerunning the source in an actual Airbyte setup gives the same number of records, though can't totally be sure about parts that seem to change like `updatedAt` timestamps... But this shouldn't block the update, IMHO.

Here's the test results (expanding on the failure, to show what worked and what didn't)

Screenshot 2023-11-28 at 16 36 19

**Comment by imrehg** [comment]
Added one more commit to set the Start Date field as date time so in the setting the helper calendar shows up nicely. 

Otherwise I've -run the checks on the commit just before that (after pulling in the changes from `master`) and things pass on my side:

Screenshot 2023-12-15 at 18 17 19

The differences in the checks were very very likely because we have a live Ashby connection that is being changed while the test is running (rather than having a dev version), so that the data previously legitimately changed between syncs. Thus on my side things look okay.

**Comment by marcosmarxm** [comment]
At the moment Airbyte doesn't have an Ashby sandbox account to run integration tests. Previously Airbyte project allowed to accept contributions without running integration but today to keep quality and ensure every connector works we restricted to this condition. I created this issue https://github.com/airbytehq/airbyte/issues/34162 and need to wait until the team can create the sandbox account.

If you need to run these changes in your project you can build the changes locally and uses it: https://docs.airbyte.com/operator-guides/using-custom-connectors Let me know if you need any further assistance here.

**Comment by marcosmarxm** [comment]
@imrehg unfortunately our team wasn't able to get an Ashby account to run integration tests.
Today to ensure connectors quality all contributions need to have integration tests passing.
I'm going to try again and see if I can get any update. If not possible I'll need to close the contribution and archive the connector in the future.

**Comment by imrehg** [comment]
Hey @marcosmarxm, I'm also attempting to stress this case through our Ashby contact to provide some support for you on this (if our words as technically inclined clients has some weight:). I do understand your processes, though, so appreciate the context you've given!

**Comment by natikgadzhi** [comment]
@imrehg any chance you can push for a test account for us? 

I looked up the usage of Ashby connector, seems rather low, so even if things go wrong, blast radius is not big. @marcosmarxm, I think it's tolerable to ship once we're happy. 

@imrehg, would you be willing to help me with a few housekeeping improvements before we move forward, though? We're stripping dockerfiles, moving to Poetry, and inlining specs and schemas into the yaml. 

I can get you an example video of what to do and a guide, but there will be a bit of work, and I'd need to rely on you for at least some smoke testing on an actual connection you have. Deal?

**Comment by imrehg** [comment]
> any chance you can push for a test account for us?

Yup, that's the plan, currently working through that with our internal Talent team who own that connection.

> would you be willing to help me with a few housekeeping improvements before we move forward, though? We're stripping dockerfiles, moving to Poetry, and inlining specs and schemas into the yaml.
>
> I can get you an example video of what to do and a guide, but there will be a bit of work, and I'd need to rely on you for at least some smoke testing on an actual connection you have. Deal?

Sounds like a good deal to me. Yup, the video / guide would be good, to know what to cover, and happy to smoke-test things.

**Comment by natikgadzhi** [comment]
@imrehg I've merged the connector cleanup PR, so this should now be unblocked. I'll trigger the CI run, and see what's going on. As long as static checks are fine, I'm inclined to merge this. 

It might need a version bump tho.

**Comment by imrehg** [comment]
@natikgadzhi hey, sorry for the delay on my part so far!

Does the above mean that the changes you've mentioned are no longer needed to be done here? Or that they can be done as a separate PR? Or it's was the "well it's fine now if they changes are not incoming"?

I'm still happy to do those 1-2-3-4-Profit steps you described.

**Comment by natikgadzhi** [comment]
Yep, the 1-2-3-4 profit is already done. Now we need to: 
- Format
- Add version change in `metadata.yaml` and `pyproject.toml`

**Comment by natikgadzhi** [comment]
Formatters are `airbyte-ci format fix all` locally. Slash command freaked out because of the PR from fork, because we're obviously great at it.

**Comment by imrehg** [comment]
Done the formatting, and bumped version (made a guess for 0.1.1 -> 0.2.0, but happy to adjust). How's this looking?

**Comment by imrehg** [comment]
@aaronsteers @natikgadzhi While I was using this data in one of our use case, I found that we are likely be missing one more entity ([openings](https://developers.ashbyhq.com/reference/openinglist)). Is that okay to add to the PR, or since it's seems to be close on your side I should wait for this PR to be merged & submit new changes in a follow up?

**Comment by imrehg** [comment]
@natikgadzhi I've been looking at this from my side as well, and realised that the previous changes you've mentioned:

> @imrehg I've merged the connector cleanup PR, so this should now be unblocked. I'll trigger the CI run, and see what's going on. As long as static checks are fine, I'm inclined to merge this.

seem to have resulted in all those merge conflicts (all the inlining, etc....) thus this PR is not fit to be merged, until all the changes are actually re-done on top of the current `master`. Do I get that right?

**Comment by natikgadzhi** [comment]
If conflicts are too much, a new PR could be easier, yeah. Adding new
streams would be wonderful! I’m
Happy to help.

File: docs/integrations/sources/ashby.md
Comment (marcosmarxm) [inline docs/integrations/sources/ashby.md:23]: It is missing the changelog update in the bottom.

File: airbyte-integrations/connectors/source-ashby/pyproject.toml
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-ashby/pyproject.toml:6]: Please run `poetry lock` in the connector folder to regenerate the dependency file.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
A few comments. @natikgadzhi are you going to merge this after the changes?

**Comment by imrehg** [comment]
Sorry, @marcosmarxm , not all changes are done yet, got hold up while doing the last few updates by a CDK issue. Will finish up shortly and update the info here.

**Pull request closed without merging**
